client/stats/keeper.go

```diff
@@ -70,6 +62,31 @@ func (keeper *sessionStatsKeeper) GetSessionDuration() time.Duration {
 }
 
 // MarkSessionEnd stops counting session duration
-func (keeper *sessionStatsKeeper) MarkSessionEnd() {
+func (keeper *SessionStatsKeeper) markSessionEnd() {
 	keeper.sessionStart = nil
 }
+
+// Subscribe subscribes the keeper on the bus for relevant events
+func (keeper *SessionStatsKeeper) Subscribe(bus client.EventSubscriptionKeeper) {
+	bus.Subscribe(string(connection.StatsEvent), keeper.consumeStatsEvent)
+	bus.Subscribe(string(connection.StateEvent), keeper.consumeStateEvent)
+}
+
+// Unsubscribe unsubscribes the sender from bus
+func (keeper *SessionStatsKeeper) Unsubscribe(bus client.EventSubscriptionKeeper) {
+	bus.Unsubscribe(string(connection.StatsEvent), keeper.consumeStatsEvent)
+	bus.Unsubscribe(string(connection.StateEvent), keeper.consumeStateEvent)
```

Comment:
bus.Unsubscribe returns an error, should we check it somehow?

Reply:
These will now be checked in DI.